### PR TITLE
planner: projection don't new unnecessary column

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Since name is splitted from the Column.
The project don't need to new column when the expr is just a column.
This makes it possible to merge the two project eliminating process which TiDB current has.

### What is changed and how it works?

If the expression is column. We use it directly, instead of new one.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: planner/core/rule_column_pruning.go [hunk 1]
```diff
@@ -42,13 +42,10 @@ func getUsedList(usedCols []*expression.Column, schema *expression.Schema) ([]bo
 		}
 	})
 
+	tmpSchema := expression.NewSchema(usedCols...)
 	used := make([]bool, schema.Len())
-	for _, col := range usedCols {
-		idx := schema.ColumnIndex(col)
-		if idx == -1 {
-			return nil, errors.Errorf("Can't find column %s from schema %s.", col, schema)
```
Comment (lamxTyler) [inline planner/core/rule_column_pruning.go]: Now this function is not expected to return errors? Do we need to update the function return values?

Comment (winoros) [reply planner/core/rule_column_pruning.go]: changed

File: planner/core/rule_column_pruning.go [hunk 2]
```diff
@@ -84,10 +73,7 @@ func exprHasSetVarOrSleep(expr expression.Expression) bool {
 // If any expression has SetVar function or Sleep function, we do not prune it.
 func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column) error {
```
Comment (lzmhhh123) [inline planner/core/rule_column_pruning.go]: It seems `PruneColumns` don't need to return `error`.

Comment (winoros) [reply planner/core/rule_column_pruning.go]: There's a error check in Aggregation's `PruneColumns`. In line 113.

**Pull request merged**